src/controllers/edgeconnect/controller.go

```diff
@@ -0,0 +1,124 @@
+package edgeconnect
+
+import (
+	"context"
+	"os"
+	"time"
+
+	edgeconnectv1alpha1 "github.com/Dynatrace/dynatrace-operator/src/api/v1alpha1/edgeconnect"
+	"github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube/version"
+	"github.com/Dynatrace/dynatrace-operator/src/kubeobjects"
+	"github.com/Dynatrace/dynatrace-operator/src/kubesystem"
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	defaultUpdateInterval = 30 * time.Minute
+)
+
+// Controller reconciles an EdgeConnect object
+type Controller struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the api-server
+	client            client.Client
+	apiReader         client.Reader
+	scheme            *runtime.Scheme
+	config            *rest.Config
+	operatorNamespace string
+	clusterID         string
+	versionProvider   version.ImageVersionFunc
+}
+
+func Add(mgr manager.Manager, _ string) error {
+	kubeSysUID, err := kubesystem.GetUID(mgr.GetAPIReader())
+	if err != nil {
+		return errors.WithStack(err)
+	}
```

Comment:
we don't need this as its connected to `clusterID`

Reply:
removed